packages/angular/src/lib/defaults/components/default-node/default-node.component.ts

```diff
@@ -1,49 +1,118 @@
+import { combineLatest } from 'rxjs';
+import { RxZuDiagramComponent } from '../../../diagram/diagram.component';
+import { FactoryService } from '../../../factory.service';
 
 @Component({
   selector: 'rxzu-default-node',
   templateUrl: './default-node.component.html',
   styleUrls: ['./default-node.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
+  host: {
+    '[attr.data-nodeid]': `model?.id`
+  }
 })
-export class DefaultNodeComponent {
+export class DefaultNodeComponent implements OnInit {
+  private portDiffers: IterableDiffer<PortModel>;
   @ViewChild('portsLayer', { read: ViewContainerRef, static: true })
   portsLayer!: ViewContainerRef;
   rootNode: HTMLElement;
 
   constructor(
-    @Inject(NODE_MODEL) public model: NodeModel,
+    @Host() @Optional() @Inject(MODEL) public model: NodeModel,
+    private factory: FactoryService,
+    private diagram: RxZuDiagramComponent,
     private rootEl: ElementRef,
-    private renderer: Renderer2
+    private renderer: Renderer2,
+    iterableDiffers: IterableDiffers
   ) {
     this.rootNode = this.rootEl.nativeElement;
+    console.log(model);
+    this.portDiffers = iterableDiffers.find([]).create<PortModel>((index, item) => item.id);
+  }
+
+  ngOnInit() {
     this.updateNodePosition();
-    this.model.setPainted(true);
+    this.updatePorts();
   }
 
   getPortsHost() {
     return this.portsLayer;
   }
 
-  updateNodePosition() {
-    // default position for node
-    this.renderer.setStyle(this.rootNode, 'position', 'absolute');
-    this.renderer.setStyle(this.rootNode, 'display', 'block');
-
+  updateNodePosition(): void {
     // subscribe to node coordinates
-    this.model.selectCoords().subscribe(({ x, y }) => {
+    this.model.selectCoords().pipe(takeUntil(this.model.onEntityDestroy())).subscribe(({ x, y }) => {
       this.renderer.setStyle(this.rootNode, 'left', `${x}px`);
       this.renderer.setStyle(this.rootNode, 'top', `${y}px`);
     });
   }
+
+  updatePorts(): void {
+    this.model
+      .selectPorts()
+      .pipe(
+        takeUntil(this.model.onEntityDestroy()),
+        filter(
+          (
+            ports: PortModel[] | null | undefined
+          ): ports is PortModel[] =>
+            ports !== null && ports !== undefined
+        ),
+        switchMap(ports => {
+          const changes = this.portDiffers.diff(ports);
+          console.log(changes);
+          this.applyPortChanges(changes);
+          return combineLatest(ports.map((port) => port.paintChanges().pipe(
+            pluck('isPainted'),
+            filter<boolean>(Boolean),
+            take(1))
+          ));
+        }),
+        filter((val) => val !== null),
+        mapTo(true)
+      ).subscribe(() => !this.model.getPainted() && this.model.setPainted(true));
+  }
+
+  private applyPortChanges(changes: IterableChanges<PortModel> | null): void {
+    changes?.forEachAddedItem(({ item, currentIndex, previousIndex }) => {
```

Comment:
can do `if(changes)` in the beginning of the function instead of `changes?.` every time